README.md
# LeetCode 解题笔记


## 简单 ( 59 题 )

- [1. 两数之和](./subject/简单/1.%20两数之和/twoSum.js)
- [100. 相同的树](./subject/简单/100.%20相同的树/isSameTree.js)
- [104. 二叉树的最大深度](./subject/简单/104.%20二叉树的最大深度/maxDepth.js)
- [107. 二叉树的层次遍历 II](./subject/简单/107.%20二叉树的层次遍历%20II/levelOrderBottom.js)
- [110. 平衡二叉树](./subject/简单/110.%20平衡二叉树/isBalanced.js)
- [112. 路径总和](./subject/简单/112.%20路径总和/hasPathSum.js)
- [1237. 找出给定方程的正整数解](./subject/简单/1237.%20找出给定方程的正整数解/findSolution.js)
- [13. 罗马数字转整数](./subject/简单/13.%20罗马数字转整数/romanToInt.js)
- [1337. 方正中战斗力最弱的 k 行](./subject/简单/1337.%20方正中战斗力最弱的%20k%20行/kWeakestRows.js)
- [1351. 统计有序矩阵中的负数](./subject/简单/1351.%20统计有序矩阵中的负数/countNegatives.js)
- [136. 只出现一次的数字](./subject/简单/136.%20只出现一次的数字/singleNumber.js)
- [14. 最长公共前缀](./subject/简单/14.%20最长公共前缀/longestCommonPrefix.js)
- [141. 环形链表](./subject/简单/141.%20环形链表/hasCycle.js)
- [155. 最小栈](./subject/简单/155.%20最小栈/minStack.js)
- [167. 两数之和 II - 输入有序数组](./subject/简单/167.%20两数之和%20II%20-%20输入有序数组/twoSum.js)
- [169. 多数元素](./subject/简单/169.%20多数元素/majorityElement.js)
- [189. 旋转数组](./subject/简单/189.%20旋转数组/rotate.js)
- [20. 有效的括号](./subject/简单/20.%20有效的括号/isValid.js)
- [21. 合并两个有序链表](./subject/简单/21.%20合并两个有序链表/mergeTwoLists.js)
- [217. 存在重复元素](./subject/简单/217.%20存在重复元素/containsDuplicate.js)
- [219. 存在重复元素 II](./subject/简单/219.%20存在重复元素%20II/containsNearbyDuplicate.js)
- [258. 各位相加](./subject/简单/258.%20各位相加/addDigits.js)
- [26. 删除排序数组中的重复项](./subject/简单/26.%20删除排序数组中的重复项/removeDuplicates.js)
- [268. 缺失数字](./subject/简单/268.%20缺失数字/missingNumber.js)
- [27. 移除元素](./subject/简单/27.%20移除元素/removeElement.go)
- [278. 第一个错误的版本](./subject/简单/278.%20第一个错误的版本/solution.js)
- [28. 实现 strStr()](./subject/简单/28.%20实现%20strStr()/strStr.js)
- [283. 移动零](./subject/简单/283.%20移动零/moveZeroes.js)
- [299. 猜数字游戏](./subject/简单/299.%20猜数字游戏/getHint.js)
- [303. 区域和检索-数组不可变](./subject/简单/303.%20区域和检索-数组不可变/sumRange.js)
- [326. 3的幂](./subject/简单/326.%203的幂/isPowerOfThree.js)
- [349. 两个数组的交集](./subject/简单/349.%20两个数组的交集/intersection.js)
- [35. 搜索插入位置](./subject/简单/35.%20搜索插入位置/searchInsert.js)
- [350. 两个数组的交集 II](./subject/简单/350.%20两个数组的交集%20II/intersect.js)
- [367. 有效的完全平方数](./subject/简单/367.%20有效的完全平方数/isPerfectSquare.js)
- [374. 猜数字大小](./subject/简单/374.%20猜数字大小/guessNumber.js)
- [389. 找不同](./subject/简单/389.%20找不同/findTheDifference.js)
- [392. 判断子序列](./subject/简单/392.%20判断子序列/isSubsequence.js)
- [404. 左叶子之和](./subject/简单/404.%20左叶子之和/sumOfLeftLeaves.js)
- [405. 数字转换为十六进制数](./subject/简单/405.%20数字转换为十六进制数/toHex.js)
- [414. 第三大的数](./subject/简单/414.%20第三大的数/thirdMax.js)
- [441. 排列硬币](./subject/简单/441.%20排列硬币/arrangeCoins.js)
- [475. 供暖器](./subject/简单/475.%20供暖器/findRadius.js)
- [485. 最大连续 1 的个数](./subject/简单/485.%20最大连续%201%20的个数/findMaxConsecutiveOnes.js)
- [509. 斐波那契数](./subject/简单/509.%20斐波那契数/fib.js)
- [555. N 叉树的最大深度](./subject/简单/555.%20N%20叉树的最大深度/maxDepth.mjs)
- [58. 最后一个单词的长度](./subject/简单/58.%20最后一个单词的长度/lengthOfLastWord.js)
- [589. N 叉树的前序遍历](./subject/简单/589.%20N%20叉树的前序遍历/preorder.js)
- [590. N叉树的后序遍历](./subject/简单/590.%20N叉树的后序遍历/postorder.js)
- [66. 加一](./subject/简单/66.%20加一/plusOne.js)
- [69. x的平方根](./subject/简单/69.%20x的平方根/mySqrt.js)
- [7. 整数反转](./subject/简单/7.%20整数反转/reverse.js)
- [70. 爬楼梯](./subject/简单/70.%20爬楼梯/climbStairs.js)
- [704. 二分查找](./subject/简单/704.%20二分查找/search.js)
- [744. 寻找比目标字母大的最小字母](./subject/简单/744.%20寻找比目标字母大的最小字母/nextGreatestLetter.js)
- [83. 删除排序链表中的重复元素](./subject/简单/83.%20删除排序链表中的重复元素/deleteDuplicates.js)
- [852. 山脉的峰顶索引](./subject/简单/852.%20山脉的峰顶索引/peakIndexInMountainArray.js)
- [88. 合并两个有序数组](./subject/简单/88.%20合并两个有序数组/merge.js)
- [9. 回文数](./subject/简单/9.%20回文数/isPalindrome.js)

## 中等 ( 16 题 )

- [102. 二叉树的层次遍历](./subject/中等/102.%20二叉树的层次遍历/levelOrder.js)
- [105. 从前序和中序遍历序列构建二叉树](./subject/中等/105.%20从前序和中序遍历序列构建二叉树/buildTree.js)
- [106. 从中序和后序遍历序列构建二叉树](./subject/中等/106.%20从中序和后序遍历序列构建二叉树/buildTree.js)
- [12. 整数转罗马数字](./subject/中等/12.%20整数转罗马数字/intToRoman.js)
- [144. 二叉树的前序遍历](./subject/中等/144.%20二叉树的前序遍历/preorderTraversal.js)
- [15. 三数之和](./subject/中等/15.%20三数之和/threeSum.js)
- [153. 寻找旋转排序数组中的最小值](./subject/中等/153.%20寻找旋转排序数组中的最小值/findMin.js)
- [16. 最接近的三数之和](./subject/中等/16.%20最接近的三数之和/threeSumClosest.js)
- [2. 两数相加](./subject/中等/2.%20两数相加/addTwoNumbers.js)
- [33. 搜索旋转排序数组](./subject/中等/33.%20搜索旋转排序数组/search.js)
- [34. 在排序数组中查找元素的第一个和最后一个位置](./subject/中等/34.%20在排序数组中查找元素的第一个和最后一个位置/searchRange.js)
- [429. N叉树的层序遍历](./subject/中等/429.%20N叉树的层序遍历/levelOrder.js)
- [655. 输出二叉树](./subject/中等/655.%20输出二叉树/printTree.js)
- [8. 字符串转换整数](./subject/中等/8.%20字符串转换整数/atoi.js)
- [889. 根据前序和后序遍历二叉树](./subject/中等/889.%20根据前序和后序遍历二叉树/constructFromPrePost.js)
- [94. 二叉树的中序遍历](./subject/中等/94.%20二叉树的中序遍历/inorderTraversal.js)

## 困难 ( 04 题 )

- [145. 二叉树的后序遍历](./subject/困难/145.%20二叉树的后序遍历/postorderTraversal.js)
- [154. 寻找旋转数组中的最小值 II](./subject/困难/154.%20寻找旋转数组中的最小值%20II/findMin.js)
- [25. k 个一组翻转链表](./subject/困难/25.%20k%20个一组翻转链表/reverseKGroup.js)
- [4. 寻找两个有序数组的中位数](./subject/困难/4.%20寻找两个有序数组的中位数/findMedianSortedArrays.js)

## 面试题 ( 02 题 )

- [08.03. 魔术索引](./subject/面试题/08.03.%20魔术索引/findMagicIndex.js)
- [10.05. 稀疏数组搜索](./subject/面试题/10.05.%20稀疏数组搜索/findString.js)

## 剑指Offer ( 08 题 )

- [11. 旋转数组的最小数字](./subject/剑指Offer/11.%20旋转数组的最小数字/minArray.js)
- [25. 合并两个排序的链表](./subject/剑指Offer/25.%20合并两个排序的链表/mergeTwoLists.js)
- [32-I. 从上到下打印二叉树](./subject/剑指Offer/32-I.%20从上到下打印二叉树/levelOrder.js)
- [32-II. 从上到下打印二叉树 II](./subject/剑指Offer/32-II.%20从上到下打印二叉树%20II/levelOrder.mjs)
- [32-III. 从上到下打印二叉树 III](./subject/剑指Offer/32-III.%20从上到下打印二叉树%20III/levelOrder.js)
- [53-I. 在排序数组中查找数组 I](./subject/剑指Offer/53-I.%20在排序数组中查找数组%20I/search.js)
- [53-II. 0~n-1中缺失的数字](./subject/剑指Offer/53-II.%200~n-1中缺失的数字/missingNumber.js)
- [55-I. 二叉树的深度](./subject/剑指Offer/55-I.%20二叉树的深度/maxDepth.js)


CHANGELOG.md
## <small>新增 62 题(2020-08-22)</small>

**简单**

- [1. 两数之和](./subject/简单/1.%20两数之和/twoSum.js)
- [104. 二叉树的最大深度](./subject/简单/104.%20二叉树的最大深度/maxDepth.js)
- [107. 二叉树的层次遍历 II](./subject/简单/107.%20二叉树的层次遍历%20II/levelOrderBottom.js)
- [1237. 找出给定方程的正整数解](./subject/简单/1237.%20找出给定方程的正整数解/findSolution.js)
- [13. 罗马数字转整数](./subject/简单/13.%20罗马数字转整数/romanToInt.js)
- [1337. 方正中战斗力最弱的 k 行](./subject/简单/1337.%20方正中战斗力最弱的%20k%20行/kWeakestRows.js)
- [1351. 统计有序矩阵中的负数](./subject/简单/1351.%20统计有序矩阵中的负数/countNegatives.js)
- [136. 只出现一次的数字](./subject/简单/136.%20只出现一次的数字/singleNumber.js)
- [14. 最长公共前缀](./subject/简单/14.%20最长公共前缀/longestCommonPrefix.js)
- [167. 两数之和 II - 输入有序数组](./subject/简单/167.%20两数之和%20II%20-%20输入有序数组/twoSum.js)
- [20. 有效的括号](./subject/简单/20.%20有效的括号/isValid.js)
- [21. 合并两个有序链表](./subject/简单/21.%20合并两个有序链表/mergeTwoLists.js)
- [26. 删除排序数组中的重复项](./subject/简单/26.%20删除排序数组中的重复项/removeDuplicates.js)
- [27. 移除元素](./subject/简单/27.%20移除元素/removeElement.go)
- [278. 第一个错误的版本](./subject/简单/278.%20第一个错误的版本/solution.js)
- [28. 实现 strStr()](<./subject/简单/28.%20实现%20strStr()/strStr.js>)
- [349. 两个数组的交集](./subject/简单/349.%20两个数组的交集/intersection.js)
- [35. 搜索插入位置](./subject/简单/35.%20搜索插入位置/searchInsert.js)
- [350. 两个数组的交集 II](./subject/简单/350.%20两个数组的交集%20II/intersect.js)
- [367. 有效的完全平方数](./subject/简单/367.%20有效的完全平方数/isPerfectSquare.js)
- [374. 猜数字大小](./subject/简单/374.%20猜数字大小/guessNumber.js)
- [392. 判断子序列](./subject/简单/392.%20判断子序列/isSubsequence.js)
- [441. 排列硬币](./subject/简单/441.%20排列硬币/arrangeCoins.js)
- [475. 供暖器](./subject/简单/475.%20供暖器/findRadius.js)
- [555. N 叉树的最大深度](./subject/简单/555.%20N%20叉树的最大深度/maxDepth.mjs)
- [58. 最后一个单词的长度](./subject/简单/58.%20最后一个单词的长度/lengthOfLastWord.js)
- [589. N 叉树的前序遍历](./subject/简单/589.%20N%20叉树的前序遍历/preorder.js)
- [590. N 叉树的后序遍历](./subject/简单/590.%20N叉树的后序遍历/postorder.js)
- [66. 加一](./subject/简单/66.%20加一/plusOne.js)
- [69. x 的平方根](./subject/简单/69.%20x的平方根/mySqrt.js)
- [7. 整数反转](./subject/简单/7.%20整数反转/reverse.js)
- [704. 二分查找](./subject/简单/704.%20二分查找/search.js)
- [744. 寻找比目标字母大的最小字母](./subject/简单/744.%20寻找比目标字母大的最小字母/nextGreatestLetter.js)
- [852. 山脉的峰顶索引](./subject/简单/852.%20山脉的峰顶索引/peakIndexInMountainArray.js)
- [88. 合并两个有序数组](./subject/简单/88.%20合并两个有序数组/merge.js)
- [9. 回文数](./subject/简单/9.%20回文数/isPalindrome.js)

**中等**

- [102. 二叉树的层次遍历](./subject/中等/102.%20二叉树的层次遍历/levelOrder.js)
- [12. 整数转罗马数字](./subject/中等/12.%20整数转罗马数字/intToRoman.js)
- [144. 二叉树的前序遍历](./subject/中等/144.%20二叉树的前序遍历/preorderTraversal.js)
- [15. 三数之和](./subject/中等/15.%20三数之和/threeSum.js)
- [153. 寻找旋转排序数组中的最小值](./subject/中等/153.%20寻找旋转排序数组中的最小值/findMin.js)
- [16. 最接近的三数之和](./subject/中等/16.%20最接近的三数之和/threeSumClosest.js)
- [2. 两数相加](./subject/中等/2.%20两数相加/addTwoNumbers.js)
- [33. 搜索旋转排序数组](./subject/中等/33.%20搜索旋转排序数组/search.js)
- [34. 在排序数组中查找元素的第一个和最后一个位置](./subject/中等/34.%20在排序数组中查找元素的第一个和最后一个位置/searchRange.js)
- [429. N 叉树的层序遍历](./subject/中等/429.%20N叉树的层序遍历/levelOrder.js)
- [655. 输出二叉树](./subject/中等/655.%20输出二叉树/printTree.js)
- [8. 字符串转换整数](./subject/中等/8.%20字符串转换整数/atoi.js)
- [94. 二叉树的中序遍历](./subject/中等/94.%20二叉树的中序遍历/inorderTraversal.js)

**困难**

- [145. 二叉树的后序遍历](./subject/困难/145.%20二叉树的后序遍历/postorderTraversal.js)
- [154. 寻找旋转数组中的最小值 II](./subject/困难/154.%20寻找旋转数组中的最小值%20II/findMin.js)
- [25. k 个一组翻转链表](./subject/困难/25.%20k%20个一组翻转链表/reverseKGroup.js)
- [4. 寻找两个有序数组的中位数](./subject/困难/4.%20寻找两个有序数组的中位数/findMedianSortedArrays.js)

**面试题**

- [08.03. 魔术索引](./subject/面试题/08.03.%20魔术索引/findMagicIndex.js)
- [10.05. 稀疏数组搜索](./subject/面试题/10.05.%20稀疏数组搜索/findString.js)

**剑指 Offer**

- [11. 旋转数组的最小数字](./subject/剑指Offer/11.%20旋转数组的最小数字/minArray.js)
- [32-I. 从上到下打印二叉树](./subject/剑指Offer/32-I.%20从上到下打印二叉树/levelOrder.js)
- [32-II. 从上到下打印二叉树 II](./subject/剑指Offer/32-II.%20从上到下打印二叉树%20II/levelOrder.mjs)
- [32-III. 从上到下打印二叉树 III](./subject/剑指Offer/32-III.%20从上到下打印二叉树%20III/levelOrder.js)
- [53-I. 在排序数组中查找数组 I](./subject/剑指Offer/53-I.%20在排序数组中查找数组%20I/search.js)
- [53-II. 0~n-1 中缺失的数字](./subject/剑指Offer/53-II.%200~n-1中缺失的数字/missingNumber.js)
- [55-I. 二叉树的深度](./subject/剑指Offer/55-I.%20二叉树的深度/maxDepth.js)

## <small>新增 4 题(2020-08-23)</small>

**简单**
- [189. 旋转数组](./subject/简单/189.%20旋转数组/rotate.js)
- [217. 存在重复元素](./subject/简单/217.%20存在重复元素/containsDuplicate.js)
- [219. 存在重复元素 II](./subject/简单/219.%20存在重复元素%20II/containsNearbyDuplicate.js)
- [283. 移动零](./subject/简单/283.%20移动零/moveZeroes.js)

## <small>新增4题(2020-08-29)</small>


**简单**


**中等**

- [105. 从前序和中序遍历序列构建二叉树](./subject/中等/105.%20从前序和中序遍历序列构建二叉树/buildTree.js)
- [106. 从中序和后序遍历序列构建二叉树](./subject/中等/106.%20从中序和后序遍历序列构建二叉树/buildTree.js)
- [889. 根据前序和后序遍历二叉树](./subject/中等/889.%20根据前序和后序遍历二叉树/constructFromPrePost.js)

**困难**


**面试题**


**剑指Offer**

- [25. 合并两个排序的链表](./subject/剑指Offer/25.%20合并两个排序的链表/mergeTwoLists.js)

## <small>新增7题(2020-08-30)</small>


**简单**

- [258. 各位相加](./subject/简单/258.%20各位相加/addDigits.js)
- [268. 缺失数字](./subject/简单/268.%20缺失数字/missingNumber.js)
- [299. 猜数字游戏](./subject/简单/299.%20猜数字游戏/getHint.js)
- [303. 区域和检索-数组不可变](./subject/简单/303.%20区域和检索-数组不可变/sumRange.js)
- [326. 3的幂](./subject/简单/326.%203的幂/isPowerOfThree.js)
- [414. 第三大的数](./subject/简单/414.%20第三大的数/thirdMax.js)
- [485. 最大连续 1 的个数](./subject/简单/485.%20最大连续%201%20的个数/findMaxConsecutiveOnes.js)

**中等**


**困难**


**面试题**


**剑指Offer**


## <small>新增3题(2020-08-31)</small>


**简单**

- [155. 最小栈](./subject/简单/155.%20最小栈/minStack.js)
- [169. 多数元素](./subject/简单/169.%20多数元素/majorityElement.js)
- [83. 删除排序链表中的重复元素](./subject/简单/83.%20删除排序链表中的重复元素/deleteDuplicates.js)

**中等**


**困难**


**面试题**


**剑指Offer**


## <small>新增1题(2020-09-02)</small>


**简单**

- [404. 左叶子之和](./subject/简单/404.%20左叶子之和/sumOfLeftLeaves.js)

**中等**


**困难**


**面试题**


**剑指Offer**


## <small>新增8题(2020-09-20)</small>


**简单**

- [100. 相同的树](./subject/简单/100.%20相同的树/isSameTree.js)
- [110. 平衡二叉树](./subject/简单/110.%20平衡二叉树/isBalanced.js)
- [112. 路径总和](./subject/简单/112.%20路径总和/hasPathSum.js)
- [141. 环形链表](./subject/简单/141.%20环形链表/hasCycle.js)
- [389. 找不同](./subject/简单/389.%20找不同/findTheDifference.js)

update-data.js
export default{
  "topic": [
    {
      "title": "简单",
      "children": [
        "[1. 两数之和](./subject/简单/1.%20两数之和/twoSum.js)",
        "[100. 相同的树](./subject/简单/100.%20相同的树/isSameTree.js)",
        "[104. 二叉树的最大深度](./subject/简单/104.%20二叉树的最大深度/maxDepth.js)",
        "[107. 二叉树的层次遍历 II](./subject/简单/107.%20二叉树的层次遍历%20II/levelOrderBottom.js)",
        "[110. 平衡二叉树](./subject/简单/110.%20平衡二叉树/isBalanced.js)",
        "[112. 路径总和](./subject/简单/112.%20路径总和/hasPathSum.js)",
        "[1237. 找出给定方程的正整数解](./subject/简单/1237.%20找出给定方程的正整数解/findSolution.js)",
        "[13. 罗马数字转整数](./subject/简单/13.%20罗马数字转整数/romanToInt.js)",
        "[1337. 方正中战斗力最弱的 k 行](./subject/简单/1337.%20方正中战斗力最弱的%20k%20行/kWeakestRows.js)",
        "[1351. 统计有序矩阵中的负数](./subject/简单/1351.%20统计有序矩阵中的负数/countNegatives.js)",
        "[136. 只出现一次的数字](./subject/简单/136.%20只出现一次的数字/singleNumber.js)",
        "[14. 最长公共前缀](./subject/简单/14.%20最长公共前缀/longestCommonPrefix.js)",
        "[141. 环形链表](./subject/简单/141.%20环形链表/hasCycle.js)",
        "[155. 最小栈](./subject/简单/155.%20最小栈/minStack.js)",
        "[167. 两数之和 II - 输入有序数组](./subject/简单/167.%20两数之和%20II%20-%20输入有序数组/twoSum.js)",
        "[169. 多数元素](./subject/简单/169.%20多数元素/majorityElement.js)",
        "[189. 旋转数组](./subject/简单/189.%20旋转数组/rotate.js)",
        "[20. 有效的括号](./subject/简单/20.%20有效的括号/isValid.js)",
        "[21. 合并两个有序链表](./subject/简单/21.%20合并两个有序链表/mergeTwoLists.js)",
        "[217. 存在重复元素](./subject/简单/217.%20存在重复元素/containsDuplicate.js)",
        "[219. 存在重复元素 II](./subject/简单/219.%20存在重复元素%20II/containsNearbyDuplicate.js)",
        "[258. 各位相加](./subject/简单/258.%20各位相加/addDigits.js)",
        "[26. 删除排序数组中的重复项](./subject/简单/26.%20删除排序数组中的重复项/removeDuplicates.js)",
        "[268. 缺失数字](./subject/简单/268.%20缺失数字/missingNumber.js)",
        "[27. 移除元素](./subject/简单/27.%20移除元素/removeElement.go)",
        "[278. 第一个错误的版本](./subject/简单/278.%20第一个错误的版本/solution.js)",
        "[28. 实现 strStr()](./subject/简单/28.%20实现%20strStr()/strStr.js)",
        "[283. 移动零](./subject/简单/283.%20移动零/moveZeroes.js)",
        "[299. 猜数字游戏](./subject/简单/299.%20猜数字游戏/getHint.js)",
        "[303. 区域和检索-数组不可变](./subject/简单/303.%20区域和检索-数组不可变/sumRange.js)",
        "[326. 3的幂](./subject/简单/326.%203的幂/isPowerOfThree.js)",
        "[349. 两个数组的交集](./subject/简单/349.%20两个数组的交集/intersection.js)",
        "[35. 搜索插入位置](./subject/简单/35.%20搜索插入位置/searchInsert.js)",
        "[350. 两个数组的交集 II](./subject/简单/350.%20两个数组的交集%20II/intersect.js)",
        "[367. 有效的完全平方数](./subject/简单/367.%20有效的完全平方数/isPerfectSquare.js)",
        "[374. 猜数字大小](./subject/简单/374.%20猜数字大小/guessNumber.js)",
        "[389. 找不同](./subject/简单/389.%20找不同/findTheDifference.js)",
        "[392. 判断子序列](./subject/简单/392.%20判断子序列/isSubsequence.js)",
        "[404. 左叶子之和](./subject/简单/404.%20左叶子之和/sumOfLeftLeaves.js)",
        "[405. 数字转换为十六进制数](./subject/简单/405.%20数字转换为十六进制数/toHex.js)",
        "[414. 第三大的数](./subject/简单/414.%20第三大的数/thirdMax.js)",
        "[441. 排列硬币](./subject/简单/441.%20排列硬币/arrangeCoins.js)",
        "[475. 供暖器](./subject/简单/475.%20供暖器/findRadius.js)",
        "[485. 最大连续 1 的个数](./subject/简单/485.%20最大连续%201%20的个数/findMaxConsecutiveOnes.js)",
        "[509. 斐波那契数](./subject/简单/509.%20斐波那契数/fib.js)",
        "[555. N 叉树的最大深度](./subject/简单/555.%20N%20叉树的最大深度/maxDepth.mjs)",
        "[58. 最后一个单词的长度](./subject/简单/58.%20最后一个单词的长度/lengthOfLastWord.js)",
        "[589. N 叉树的前序遍历](./subject/简单/589.%20N%20叉树的前序遍历/preorder.js)",
        "[590. N叉树的后序遍历](./subject/简单/590.%20N叉树的后序遍历/postorder.js)",
        "[66. 加一](./subject/简单/66.%20加一/plusOne.js)",
        "[69. x的平方根](./subject/简单/69.%20x的平方根/mySqrt.js)",
        "[7. 整数反转](./subject/简单/7.%20整数反转/reverse.js)",
        "[70. 爬楼梯](./subject/简单/70.%20爬楼梯/climbStairs.js)",
        "[704. 二分查找](./subject/简单/704.%20二分查找/search.js)",
        "[744. 寻找比目标字母大的最小字母](./subject/简单/744.%20寻找比目标字母大的最小字母/nextGreatestLetter.js)",
        "[83. 删除排序链表中的重复元素](./subject/简单/83.%20删除排序链表中的重复元素/deleteDuplicates.js)",
        "[852. 山脉的峰顶索引](./subject/简单/852.%20山脉的峰顶索引/peakIndexInMountainArray.js)",
        "[88. 合并两个有序数组](./subject/简单/88.%20合并两个有序数组/merge.js)",
        "[9. 回文数](./subject/简单/9.%20回文数/isPalindrome.js)"
      ]
    },
    {
      "title": "中等",
      "children": [
        "[102. 二叉树的层次遍历](./subject/中等/102.%20二叉树的层次遍历/levelOrder.js)",
        "[105. 从前序和中序遍历序列构建二叉树](./subject/中等/105.%20从前序和中序遍历序列构建二叉树/buildTree.js)",
        "[106. 从中序和后序遍历序列构建二叉树](./subject/中等/106.%20从中序和后序遍历序列构建二叉树/buildTree.js)",
        "[12. 整数转罗马数字](./subject/中等/12.%20整数转罗马数字/intToRoman.js)",
        "[144. 二叉树的前序遍历](./subject/中等/144.%20二叉树的前序遍历/preorderTraversal.js)",
        "[15. 三数之和](./subject/中等/15.%20三数之和/threeSum.js)",
        "[153. 寻找旋转排序数组中的最小值](./subject/中等/153.%20寻找旋转排序数组中的最小值/findMin.js)",
        "[16. 最接近的三数之和](./subject/中等/16.%20最接近的三数之和/threeSumClosest.js)",
        "[2. 两数相加](./subject/中等/2.%20两数相加/addTwoNumbers.js)",
        "[33. 搜索旋转排序数组](./subject/中等/33.%20搜索旋转排序数组/search.js)",
        "[34. 在排序数组中查找元素的第一个和最后一个位置](./subject/中等/34.%20在排序数组中查找元素的第一个和最后一个位置/searchRange.js)",
        "[429. N叉树的层序遍历](./subject/中等/429.%20N叉树的层序遍历/levelOrder.js)",
        "[655. 输出二叉树](./subject/中等/655.%20输出二叉树/printTree.js)",
        "[8. 字符串转换整数](./subject/中等/8.%20字符串转换整数/atoi.js)",
        "[889. 根据前序和后序遍历二叉树](./subject/中等/889.%20根据前序和后序遍历二叉树/constructFromPrePost.js)",
        "[94. 二叉树的中序遍历](./subject/中等/94.%20二叉树的中序遍历/inorderTraversal.js)"
      ]
    },
    {
      "title": "困难",
      "children": [
        "[145. 二叉树的后序遍历](./subject/困难/145.%20二叉树的后序遍历/postorderTraversal.js)",
        "[154. 寻找旋转数组中的最小值 II](./subject/困难/154.%20寻找旋转数组中的最小值%20II/findMin.js)",
        "[25. k 个一组翻转链表](./subject/困难/25.%20k%20个一组翻转链表/reverseKGroup.js)",
        "[4. 寻找两个有序数组的中位数](./subject/困难/4.%20寻找两个有序数组的中位数/findMedianSortedArrays.js)"
      ]
    },
    {
      "title": "面试题",
      "children": [
        "[08.03. 魔术索引](./subject/面试题/08.03.%20魔术索引/findMagicIndex.js)",
        "[10.05. 稀疏数组搜索](./subject/面试题/10.05.%20稀疏数组搜索/findString.js)"
      ]
    },
    {
      "title": "剑指Offer",
      "children": [
        "[11. 旋转数组的最小数字](./subject/剑指Offer/11.%20旋转数组的最小数字/minArray.js)",
        "[25. 合并两个排序的链表](./subject/剑指Offer/25.%20合并两个排序的链表/mergeTwoLists.js)",
        "[32-I. 从上到下打印二叉树](./subject/剑指Offer/32-I.%20从上到下打印二叉树/levelOrder.js)",
        "[32-II. 从上到下打印二叉树 II](./subject/剑指Offer/32-II.%20从上到下打印二叉树%20II/levelOrder.mjs)",
        "[32-III. 从上到下打印二叉树 III](./subject/剑指Offer/32-III.%20从上到下打印二叉树%20III/levelOrder.js)",
        "[53-I. 在排序数组中查找数组 I](./subject/剑指Offer/53-I.%20在排序数组中查找数组%20I/search.js)",
        "[53-II. 0~n-1中缺失的数字](./subject/剑指Offer/53-II.%200~n-1中缺失的数字/missingNumber.js)",
        "[55-I. 二叉树的深度](./subject/剑指Offer/55-I.%20二叉树的深度/maxDepth.js)"
      ]
    }
  ],
  "size": 89
}